### PR TITLE
Fixes #4334: Update to react-native 0.55.4

### LINKED ISSUE #4334: Update to react-native 0.55.4
Update react-native version to 0.55.4

Main advantages:
- performance improvement (more features for native drivers for animations)
- letter spacing in android
- keyboard fix for certain devices
- bug fixes

To check:
- no regressions

Bounty might be increased if it turns out that the update required more changes than expected.

### PR DESCRIPTION
**Pull request by pacamara**
Fixes #4334: Update to react-native 0.55.4

### Summary:

Upgrade to react-native 0.55.4.

Have tested on Android but not yet on IOS.

### Steps to test:

General smoke test of status-react.

status: ready

### CODE REVIEW

**Review by jeluard — APPROVED** [review]

**Comment by yenda** [comment]
Hi @pacamara thanks for the PR, could you provide screens before/after of the chat list screen, wallet and settings as well as one chat ? Design team needs to validate that the upgrade didn't modify the UI.

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/4ibmn2 for branch PR-4376

<blockquote><div><strong><a href="https://i.diawi.com/4ibmn2">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by yenda** [comment]
@pacamara are you on our Riot chat btw ?

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/dqcGA6 for branch PR-4376

<blockquote><div><strong><a href="https://i.diawi.com/dqcGA6">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by pacamara** [comment]
@yenda Sure will do the screenshots! But have realized that `npm install` is hanging on a fresh checkout of this branch, so there may be other issues, digging into that. I'm pacamara on riot chat.

**Comment by pacamara** [comment]
@yenda OK that was a local build issue with fresh checkouts that I'd forgotten about, not related to these changes :)   Android screenshots attached, from fresh checkouts of `develop` and `develop_issue4334_upgradeToReactNative0.55.4`. npm cache cleaned before building each one. :wine_glass:   

**Chat list before/after:**
chat_0 53 3
chat_0 55 4

Wallet before/after:
wallet_0 53 3
wallet_0 55 4

Settings before/after:
settings_0 53 3
settings_0 55 4

Chat before/after:
chat_0 53 3
chat_0 55 4

Seems there is a problem in 0.55.4 showing the fiat value of assets.

**Comment by yenda** [comment]
@pacamara awesome thanks for the screenshot. I think it is better if you fix the bug with the fiat value of assets before moving to testing because it won't pass QA

**Comment by pacamara** [comment]
@yenda Sure, i'm working on it  :+1:

**Comment by jeluard** [comment]
@pacamara Can you resolve the conflict? Sorry about that!

**Comment by pacamara** [comment]
Hi @jeluard , sure will do!

@yenda : The missing fiat prices are because *all* HTTP GET calls are failing: the `rn-dependencies/fetch` in `http/get` to `cryptocompare.com` returns a blank `bodyText`. Perhaps because the implementation of `fetch` has changed in RN 0.55.4? (since bodyText is private). That's how far I've got with it, won't be able to look at this again until tomorrow now, unfortunately. :beers:

**Comment by pacamara** [comment]
@yenda @jeluard Hi there! Ok the solution is to use `response.text`, have pushed a fix for this to `http.cljs` which works. Fiat values of assets now appear again in the wallet. :wine_glass:

**Comment by jeluard** [comment]
@pacamara Awesome! It looks like you have to rebase, sorry!

**Comment by pacamara** [comment]
@jeluard Yes sorry! working on that, will push again  :)

**Comment by pacamara** [comment]
@jeluard OK rebased and pushed, cheers! :beers:

File: src/status_im/utils/http.cljs [hunk 1]
```diff
@@ -38,11 +38,14 @@
...
-                             ok?)]
-                  [(.-_bodyText response) ok?'])))
+                (->
+                 (.text response)
+                 (.then (fn [x]
+                          (let [ok?  (.-ok response)
+                                ok?' (if valid-response?
+                                       (and ok? (valid-response? response))
+                                       ok?)]
+                            [x ok?']))))))
```
Comment (dmitryn) [inline src/status_im/utils/http.cljs]: I'd assume that `x` is response body, maybe we can use another naming?

**Review by dmitryn — APPROVED** [review]

**Comment by pacamara** [comment]
>>I'd assume that x is response body, maybe we can use another naming?

@dmitryn Yes that's right, sure, sorted! :beers:

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/xbY2tA for branch PR-4376

<blockquote><div><strong><a href="https://i.diawi.com/xbY2tA">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/vVoEcT for branch PR-4376

<blockquote><div><strong><a href="https://i.diawi.com/vVoEcT">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by yenda** [comment]
Good job @pacamara thanks! The PR will now go into QA (testing), it might take a few days because we are currently including a lot of changes that needs to be in beta and they have priority.

**Review by yenda — APPROVED** [review]

**Comment by pacamara** [comment]
@yenda :cool: Got it thanks :)

**Comment by rasom** [comment]
apk uploaded to https://i.diawi.com/v3b24Q for branch PR-4376

<blockquote><div><strong><a href="https://i.diawi.com/v3b24Q">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by rasom** [comment]
ipa uploaded to https://i.diawi.com/QhSau2 for branch PR-4376

<blockquote><div><strong><a href="https://i.diawi.com/QhSau2">Install - Diawi - Development and In-house Apps Wireless Installation</a></strong></div></blockquote>

**Comment by asemiankevich** [comment]
I've checked screens look same as in nightly now, also run a smoke test to verify the functionality. Looks good to me.

**Pull request merged**